packages/x-internals/src/warning/slotWarning.ts

```diff
@@ -0,0 +1,14 @@
+import { warnOnce } from './warning';
+
+export function checkSlot(slotName: string, slot: Function) {
+  if (process.env.NODE_ENV !== 'production') {
+    if (slot.name !== undefined && slot.name === slotName) {
```

Comment:
Maybe give some context on why this check tells us if this is a render function

Reply:
Added in https://github.com/mui/mui-x/pull/14748/files#diff-bcce1cbc47a034db867e72300a0f523b9be6efb8c0cfefd695ef848f97965ff2R4-R10